org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4Solution.java

```diff
@@ -313,24 +323,31 @@
         TupleSet stringBounds = factory.noneOf(1);
         final TupleSet next = factory.noneOf(2);
         int min = min(), max = max();
-        if (max >= min)
-            for (int i = min; i <= max; i++) { // Safe since we know 1 <=
-                                              // bitwidth <= 30
-                Tuple ii = factory.tuple("" + i);
+        if (max >= min){
+            ConcurrentSkipListSet<Integer> ints = new ConcurrentSkipListSet<Integer>();
+            for(String a: atoms){
+                try{
+                    ints.add(Integer.valueOf(a));
+                }catch(NumberFormatException e){}
```

Comment:
Catching `NumberFormatException` and ignoring it does not seem safe.  If this exception can happen due to a user error, I'd rather it blows up all the way to the top

Reply:
it's for filtering integers. non-integers could be safely ignored.